exercises/rail-fence-cipher/.meta/src/reference/java/RailFenceCipher.java

```diff
@@ -0,0 +1,74 @@
+import java.util.ArrayList;
+import java.util.List;
+
+class RailFenceCipher {
+
+    private int noOfRails;
+
+    RailFenceCipher(int noOfRails) {
+        this.noOfRails = noOfRails;
+    }
+
+    String getEncryptedData(String plainText) {
+        List<String> railFence = new ArrayList<>();
+        for (int i = 0; i < noOfRails; i++) {
+            railFence.add("");
+        }
+
+        int number = 0, increment = 1;
+        for (char c : plainText.toCharArray()) {
+            if (number + increment == noOfRails) {
+                increment = -1;
+            } else if (number + increment == -1) {
+                increment = 1;
+            }
+            String temp = railFence.get(number);
+            railFence.remove(number);
+            railFence.add(number, temp + c);
+            number += increment;
+        }
+        String buffer = "";
+        for (String s : railFence) {
+            buffer = buffer.concat(s);
+        }
+        return buffer;
+    }
+
+    String getDecryptedData(String cipherText) {
+        String decrypted;
+        ArrayList<ArrayList<Integer>> railFence = new ArrayList<>(noOfRails);
+
+        for (int i = 0; i < noOfRails; ++i) {
+            railFence.add(new ArrayList<>(cipherText.length()));
+        }
+
+        int rowIncrementFactor = 0, step = 1;
+
+        for (int i = 0; i < cipherText.length(); ++i) {
```

Comment:
Again, this is a bit hard to follow. Say you have n rails, then you want to go through the ciphertext and add each letter to the correct rail. You can find the correct rail by `i % n`. If `i % n` is 0 then you add it to the first rail, if `i % n` is 1 then you add it to the second rail. Does that make sense?